packages/x-components/src/x-modules/search/components/__tests__/banners-list.spec.ts

```diff
@@ -0,0 +1,129 @@
+import { Banner } from '@empathyco/x-types';
+import { createLocalVue, mount, Wrapper } from '@vue/test-utils';
+import Vue, { VueConstructor, ComponentOptions } from 'vue';
+import Vuex, { Store } from 'vuex';
+import BaseGrid from '../../../../components/base-grid.vue';
+import { getXComponentXModuleName, isXComponent } from '../../../../components/x-component.utils';
+import { XPlugin } from '../../../../plugins/x-plugin';
+import { RootXStoreState } from '../../../../store/store.types';
+import { DeepPartial, Dictionary } from '../../../../utils/types';
+import { getDataTestSelector, installNewXPlugin } from '../../../../__tests__/utils';
+import { getBannersStub } from '../../../../__stubs__/banners-stubs.factory';
+import BannersList from '../banners-list.vue';
+import { resetXSearchStateWith } from './utils';
+
+/**
+ * Renders the `BannersList` component, exposing a basic API for testing.
+ *
+ * @param options - The options to render the component with.
+ * @returns The API for testing the `BannersList` component.
+ */
+function renderBannersList({
+  template = '<BannersList />',
+  banners = getBannersStub(),
+  components
+}: RenderBannersListOptions = {}): RendersBannerListAPI {
+  const localVue = createLocalVue();
+  localVue.use(Vuex);
+  const store = new Store<DeepPartial<RootXStoreState>>({});
+  installNewXPlugin({ store }, localVue);
+
+  XPlugin.resetInstance();
+
+  resetXSearchStateWith(store, { banners });
+  const wrapper = mount(
+    {
+      components: {
+        BannersList,
+        ...components
+      },
+      template
+    },
+    {
+      localVue,
+      store
+    }
+  );
+
+  const bannersListWrapper = wrapper.findComponent(BannersList);
+
+  return {
+    wrapper: bannersListWrapper,
+    getBanners() {
+      return banners;
+    }
+  };
+}
+
+describe('testing BannersList component', () => {
+  it('is an XComponent', () => {
+    const { wrapper } = renderBannersList();
+    expect(isXComponent(wrapper.vm)).toEqual(true);
+  });
+
+  it('has Search as XModule', () => {
+    const { wrapper } = renderBannersList();
+    expect(getXComponentXModuleName(wrapper.vm)).toEqual('search');
+  });
+
+  it('renders the banners in the state', () => {
+    const { wrapper, getBanners } = renderBannersList();
+    const bannersListItems = wrapper.findAll(getDataTestSelector('banners-list-item'));
+
+    getBanners().forEach((result, index) => {
```

Comment:
You can use `every` instead of `forEach`

```
getBanners().every((result, index) => {
  expect(bannersListItems.at(index).text()).toEqual(result.title);
});
```

Reply:
Agree! I copied and paste the tests from the results-list. I'm gonna change it here and in the promoteds and results lists.

Reply:
"recojocable". The expect function evaluates itself without returning any value. What we could to make it work would be:

```
expect(
  getBanners().every((result, index) => (
    bannersListItems.at(index).text() === result.title
  ))
).toBe(true)
```

But I found it more difficult.

Reply:
🔝